plugins/org.eclipse.emf.mwe2.language/src/org/eclipse/emf/mwe2/language/validation/IntAwareRawSuperTypes.java

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ * Copyright (c) 2018 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.emf.mwe2.language.validation;
+
+import static org.eclipse.emf.mwe2.language.mwe2.impl.JvmTypeUriFactory.*;
+
+import java.util.Set;
+
+import org.eclipse.emf.mwe2.language.mwe2.impl.JvmTypeUriFactory;
+import org.eclipse.xtext.common.types.JvmPrimitiveType;
+import org.eclipse.xtext.common.types.JvmType;
+import org.eclipse.xtext.common.types.util.Primitives;
+import org.eclipse.xtext.common.types.util.RawSuperTypes;
+
+import com.google.inject.Inject;
+
+@SuppressWarnings("restriction")
+public class IntAwareRawSuperTypes extends RawSuperTypes {
+
+	@Inject 
+	private Primitives primitives;
+	
+	@Override
+	public Set<JvmType> collect(JvmType type) {
+		Set<JvmType> result = new IntAwareImplementation().collect(type);
+		result.remove(type);
+		return result;
+	}
+	
+	protected class IntAwareImplementation extends Implementation {
```

Comment:
do we have a better name for the class?
should we port it down to xbase someday?

Reply:
I'll add a comment. Which name would you suggest?

Reply:
MweRawSupertypes?